IPython/utils/text.py

```diff
@@ -639,7 +639,7 @@ def _find_optimal(rlist, row_first=False, separator_size=2, displaywidth=80):
         if sumlength + separator_size * (ncols - 1) <= displaywidth:
             break
     return {'num_columns': ncols,
-            'optimal_separator_width': (displaywidth - sumlength) / (ncols - 1) if (ncols - 1) else 0,
+            'optimal_separator_width': (displaywidth - sumlength) // (ncols - 1) if (ncols - 1) else 0,
```

Comment:
this resulted in an optimal_separator_width of e.g. 2.5 on the doctest example for `compute_item_matrix` below which is probably not what was intended.